paho/extensions/rpc/rpc.go

```diff
@@ -68,7 +68,7 @@ func (h *Handler) Request(pb *paho.Publish) (*paho.Publish, error) {
 	pb.Properties.ResponseTopic = fmt.Sprintf("%s/responses", h.c.ClientID)
 	pb.Retain = false
 
-	_, err := h.c.Publish(context.Background(), pb)
+	_, err := h.c.Publish(ctx, pb)
 	if err != nil {
 		return nil, err
 	}
```

Comment:
Should really make the same change here as you have in AutoPaho (stop trying to receive if ctx.Done()).